monkey/monkey_island/cc/island_event_handlers/set_simulation_mode.py

```diff
@@ -0,0 +1,19 @@
+from typing import Any
+
+from monkey_island.cc.repository import ISimulationRepository
+
+
+class set_simulation_mode:
+    """
+    Callable class that sets the Island's mode
+    """
+
+    def __init__(
+        self,
+        simulation_repository: ISimulationRepository,
+    ):
+        self._simulation_repository = simulation_repository
+
+    def __call__(self, event: Any = None):
```

Comment:
See comment on naming.

If we use `mode` as the parameter name instead of `event`, we can just subscribe `simulation_repository.set_mode()` directly to the event and remove this class.